test/spec/invertedIndexSpec.js

```diff
@@ -97,77 +85,114 @@ describe('InvertedIndex', () => {
+      index.createIndex(books, 'books.json');
     });
 
     it('should return an object containing the search results', () => {
-      expect(typeof index.searchAll('alice').files).toEqual('object');
+      expect(typeof index.searchIndex('alice')).toEqual('object');
     });
-    it('object should contain an array of files that contains the search key', () => {
-      expect(index.searchAll('alice').files).toEqual(
-        ['simpleValidFile.json', 'validFile.json']
+
+    it('result should contain an array of files where the search key was found',
+    () => {
+      expect(index.searchIndex('alice').files).toEqual(
+        ['shortStories.json', 'books.json']
         );
     });
+
     it('should return valid search results', () => {
-      expect(index.searchAll('alice')[this.filename]).toEqual(
+      expect(index.searchIndex('alice seek')).toEqual(
         {
-          bookTitles: ['Alice', 'Lord of the Rings', 'Rings'],
-          allWords: ['alice'],
-          words: {
-            alice: [true, false, false],
+          files: ['shortStories.json', 'books.json'],
+          'books.json': {
+            allWords: ['alice', 'seek'],
+            words: {
+              alice: [true, false, false],
+              seek: [false, true, true]
+            }
+          },
+          'shortStories.json': {
+            allWords: ['alice'],
+            words: {
+              alice: [true, false, false]
+            }
           }
         }
       );
     });
+
     it('should return false if nothing was found', () => {
-      expect(index.searchAll('abcdefgh#')).toBeFalsy();
+      expect(index.searchIndex('abcdefgh#')).toBeFalsy();
+    });
+
+    it('should return false if searchKey is an empty string', () => {
+      expect(index.searchIndex('', filename)).toBeFalsy();
+    });
+
+    it('can search correctly when filename is passed in', () => {
+      expect(index.searchIndex('alice hahaha unusual', filename)).toEqual(
+        {
+          files: ['shortStories.json'],
+          'shortStories.json': {
+            allWords: ['alice', 'unusual'],
+            words: {
+              alice: [true, false, false],
+              unusual: [false, true, true],
+            }
+          }
+        }
+      );
     });
   });
 
-  describe('DeleteFile', () => {
+  describe('DeleteFileIndex', () => {
     it('can successfully delete a file from the index', () => {
-      expect(index.deleteFile(this.filename)).toBeTruthy();
-      expect(index[this.filename]).toEqual(undefined);
-      expect(index.files.indexOf(this.filename)).toBe(-1);
+      expect(index.deleteFileIndex(filename)).toBeTruthy();
+      expect(index.getIndex(filename)).toBeFalsy();
+      expect(index.files.indexOf(filename)).toBe(-1);
     });
+
     it('returns false if filename does not exist in the index', () => {
-      expect(index.deleteFile('abcdef.json')).toBeFalsy();
+      expect(index.deleteFileIndex('abcdef.json')).toBeFalsy();
     });
   });
 });
```

Comment:
"Ubiquitous Language," a body of knowledge that can be understood by both the developer and the customer. This ubiquitous language is used to shape and develop the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", () => {

    it("should be able to add 2 numbers together", () => {
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", () => {
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```
Please review your test paying more attention to the language used on your "it" and 'describe' functions